File: Ordenamiento/insertionSort/insertionSort.c
#include <stdio.h>

int main()
{
    /* code */

    int arr[14] = {304, 37, 12, 45, 6, 2, 11, 1, 3, 5, 7, 42, 31, 88};

    for (int i = 0; i < 14; i++)
    {
        printf("%d ", arr[i]);
    }

    for (int i = 0; i < 14; i++)
    {
        int clave = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > clave)
        {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = clave;
    }
    printf("\n");
    for (int i = 0; i < 14; i++)
    {
        printf("%d ", arr[i]);
    }
    return 0;
}

File: Ordenamiento/insertionSort/insertionSort.js
let arr = [2,1,3,34,7,4,5,6,9,64, 34, 25, 12, 22, 11, 90]
console.log(arr.length)
function insertionSort(list){
    for (let i = 1; i< list.length - 1 ;i++){
           let clave = list[i]
           let j = i-1
        while(j>=0 && list[j]> clave){
            list[j+1] = list[j]
            j= j-1
        }
        list[j+1] = clave
    }
}
insertionSort(arr)
console.log(arr)